### PR TITLE
Adding an ETag header for local packager

### PR DESCRIPTION
**Pull request by MattFoley**
This is much needed for my local development pipeline. I am downloading various JS packages in the background and running them after they've been fully downloaded.

### CODE REVIEW

File: packager/react-packager/src/Packager/Package.js [hunk 1]
```diff
@@ -12,6 +12,8 @@ var _ = require('underscore');
 var base64VLQ = require('./base64-vlq');
 var UglifyJS = require('uglify-js');
 var ModuleTransport = require('../lib/ModuleTransport');
+var crypto = require('crypto');
+var fs = require('fs');
```
Comment (ide) [inline packager/react-packager/src/Packager/Package.js]: remove, unused

**Comment by ide** [comment]
LGTM, will let @amasad do a final review and merge

**Comment by MattFoley** [comment]
Removed the unused line, thanks for the review!

**Comment by MattFoley** [comment]
Hold on this, I believe a few things need to change for the ETag to be fully functional and useful. (I'm planning on using it as a cache header)

**Comment by MattFoley** [comment]
I've updated the packager to actually respond with a 304 if the Etags are a match. This is now good to go @amasad. I would be soooo happy if this was merged to master today. :) Thanks!

File: packager/react-packager/src/Server/index.js [hunk 2]
```diff
@@ -363,7 +363,13 @@ Server.prototype.processRequest = function(req, res, next) {
           minify: options.minify,
         });
         res.setHeader('Content-Type', 'application/javascript');
-        res.end(bundleSource);
+        res.setHeader('ETag', p.calculateETag());
+        if (req.headers.etag === res.getHeader('ETag')) {
```
Comment (ide) [inline packager/react-packager/src/Server/index.js]: shouldn't this be If-None-Match on the incoming request?

**Comment by ide** [comment]
@MattFoley also could you squash your commits?

**Comment by MattFoley** [comment]
Will have this updated in just a few.

**Comment by MattFoley** [comment]
Now using `if-none-match` to follow spec appropriately and I've squashed the commits into one.

**Comment by MattFoley** [comment]
@amasad @frantic Sorry to ping you again, but is there any way one of you could give this a thumbs up/ thumbs down?

**Comment by frantic** [comment]
LGTM, but please check with @amasad

**Comment by ide** [comment]
Separately (in this commit or another PR) it would also be nice to have unit tests since broken etags are hard to notice.

**Comment by MattFoley** [comment]
About a month now with no updates on this pull request. Any status updates?

**Comment by MattFoley** [comment]
@ide now that this pull request has lived so long, (I've basically rewritten the changes this morning) how do you suggest I squash these? Should I close and reopen this PR from a different branch?

**Comment by ide** [comment]
Rebase against master and squash your commits at the same time.

**Comment by MattFoley** [comment]
No, too many issues with trying to merge in these ~10 lines of code (something like 300 commits). I'm closing this and opening a new pull request.

**Comment by MattFoley** [comment]
See this commit: https://github.com/facebook/react-native/pull/2473

**Pull request closed without merging**